extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ExpositionConfig.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.kubernetes.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class ExpositionConfig {
+
+    /**
+     * If true, the service will be exposed
+     */
+    @ConfigItem
+    boolean expose;
+
+    /**
+     * The host under which the application is going to be exposed
+     */
+    @ConfigItem
+    Optional<String> host;
+
+    /**
+     * If true, TLS will be enabled on the exposed service
+     */
+    @ConfigItem(defaultValue = "true")
+    boolean secure;
```

Comment:
Any chance we could handle the annotations added to the route while we're at it?

At the moment, I added them globally but I think it makes sense to be able to define annotations specific to the route?

Note that I'm not an OpenShift expert so better check with @geoand and @iocanel if it makes sense :).

Reply:
Can you remind me again what annotation you needed on the Route?

Reply:
Looking at `dekorate`, I'm not sure we can configure annotation for a specific type of object. But may be wrong. 
I added the `kubernetes.io/tls-acme` annotation to enable tls. But as mentionned by @gsmet in an other comment, enabling tls may have more consequences than just adding an annotation (like certificate, key, ...)

Reply:
So I needed an annotation for defining the `secure` route so it's mostly covered now :).

Reply:
It does make sense!
And dekorate does support adding Labels and Annotations to specific resource types. See: https://github.com/dekorateio/dekorate/pull/644